src/Tools/ExternalAccess/Razor/Features/IRazorLSPDynamicFileInfoProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor.Features;
+
+internal interface IRazorLSPDynamicFileInfoProvider
```

Comment:
"LSP" in the name here seems odd. Isn't this just an `IRazorDynamicFileInfoProvider`, and the implementation happens to use LSP in VS Code, and not in VS?

The only thing that is "LSP-y" about this seems to be the `IRazorClientLanguageServerManager` bit on the initialize method, so maybe my above comment about passing in a `IRazorServiceProvider` would resolve that?

Reply:
Yea, let me rethink this. The main thing is that the LSP portion = needs a way to communicate using `IRazorClientLanguageServerManager`. Let me see if I can get rid of the initialize method and change the name